api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -229,7 +230,8 @@ public void deleteWorkspace(DbWorkspace dbWorkspace) {
         dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName());
     dbWorkspace.setWorkspaceActiveStatusEnum(WorkspaceActiveStatus.DELETED);
     dbWorkspace = workspaceDao.saveWithLastModified(dbWorkspace);
-    maybeDeleteRecentWorkspace(dbWorkspace.getWorkspaceId());
+    userRecentWorkspaceDao.deleteByUserIdAndWorkspaceIdIn(
```

Comment:
This entire DB interaction is arguably unnecessary. The recent workspaces API already needs to be resilient to workspace deletion. Note that any number of users could have this workspace sitting in their `user_recent_workspace` table, but we're only deleting it for the caller. IIRC the recent workspaces API will lazily delete these oprhaned entries.

I'm fine with this change, or with just removing it. If you keep this here though, maybe add a comment as to why we're deleting, and the caveat that this is not exhaustive.

Reply:
right.. what if I just delete all recent workspaces linked to the deleted workspace then? made a commit w/ that change @calbach

Reply:
This is probably OK. I didn't suggest this as an approach because I am slightly worried about a model where an action from one user has a cascading effect across a potentially large number of other users (e.g. potentially causing conflicts with other users' own recent workspace updates), but in practice I'm not sure it will matter.